# HG changeset patch
# User Jacob Dawid <jacob.dawid@gmail.com>
# Date 1342849005 14400
#      Sat Jul 21 01:36:45 2012 -0400
# Node ID f620ecde25663665c7d3a15af1004f247eaf752e
# Parent  aa6cba8ae88bec3a3c833ea1b8b0eed9c83ba5ba
Added support for changing the cursor.

diff --git a/libqterminal/QTerminalInterface.h b/libqterminal/QTerminalInterface.h
--- a/libqterminal/QTerminalInterface.h
+++ b/libqterminal/QTerminalInterface.h
@@ -44,16 +44,28 @@ public:
       connect (pasteAction, SIGNAL (triggered()), this, SLOT (pasteClipboard()));
     }
     virtual ~QTerminalInterface() { }
 
     virtual void setTerminalFont(const QFont& font) = 0;
     virtual void setSize(int h, int v) = 0;
     virtual void sendText(const QString& text) = 0;
 
+    enum CursorType {
+      UnderlineCursor,
+      BlockCursor,
+      IBeamCursor
+    };
+
+    virtual void setCursorType(CursorType type, bool blinking) {
+        // Provide empty default impl in order to avoid conflicts with the win impl.
+        Q_UNUSED(type);
+        Q_UNUSED(blinking);
+    }
+
 public slots:
     virtual void copyClipboard() = 0;
     virtual void pasteClipboard() = 0;
 
     virtual void handleCustomContextMenuRequested(QPoint at) {
       _contextMenu->move (mapToGlobal(at));
       _contextMenu->show ();
     }
diff --git a/libqterminal/unix/QUnixTerminalImpl.cpp b/libqterminal/unix/QUnixTerminalImpl.cpp
--- a/libqterminal/unix/QUnixTerminalImpl.cpp
+++ b/libqterminal/unix/QUnixTerminalImpl.cpp
@@ -113,16 +113,26 @@ void QUnixTerminalImpl::setSize(int h, i
     m_terminalView->setSize(h, v);
 }
 
 void QUnixTerminalImpl::sendText(const QString& text)
 {
     m_terminalModel->sendText(text);
 }
 
+void QUnixTerminalImpl::setCursorType(CursorType type, bool blinking)
+{
+    switch(type) {
+        case UnderlineCursor: m_terminalView->setKeyboardCursorShape(TerminalView::UnderlineCursor);
+        case BlockCursor: m_terminalView->setKeyboardCursorShape(TerminalView::BlockCursor);
+        case IBeamCursor: m_terminalView->setKeyboardCursorShape(TerminalView::IBeamCursor);
+    }
+    m_terminalView->setBlinkingCursor(blinking);
+}
+
 void QUnixTerminalImpl::focusInEvent(QFocusEvent *focusEvent)
 {
     Q_UNUSED(focusEvent);
     m_terminalView->updateImage();
     m_terminalView->repaint();
     m_terminalView->update();
 }
 
diff --git a/libqterminal/unix/QUnixTerminalImpl.h b/libqterminal/unix/QUnixTerminalImpl.h
--- a/libqterminal/unix/QUnixTerminalImpl.h
+++ b/libqterminal/unix/QUnixTerminalImpl.h
@@ -33,16 +33,18 @@ class QUnixTerminalImpl : public QTermin
 public:
     QUnixTerminalImpl(QWidget *parent = 0);
     virtual ~QUnixTerminalImpl();
     
     void setTerminalFont(const QFont &font); 
     void setSize(int h, int v);
     void sendText(const QString& text);
 
+    void setCursorType(CursorType type, bool blinking);
+
 public slots:
     void copyClipboard();
     void pasteClipboard();
         
 protected:
     void focusInEvent(QFocusEvent *focusEvent);
     void showEvent(QShowEvent *);
     virtual void resizeEvent(QResizeEvent *);   
